lib/shelley/test/integration/Main.hs

```diff
@@ -321,3 +337,53 @@ minSeverityFromEnv def var = lookupEnv var >>= \case
     Nothing -> pure def
     Just "" -> pure def
     Just arg -> either die pure (parseLoggingSeverity arg)
+
+-- | Pre-register a staking key for the STAKE_POOLS_JOIN_05 test.
+preregisterStakingKeysForTests :: NetworkLayer IO t b -> IO ()
+preregisterStakingKeysForTests nl = do
+    let payload = TxPayload
+            [toStakeKeyRegCert rewardPub]
+            (const mempty)
+    let keystore = const (Just (xprv, mempty))
+    let txIn = initialFundsPseudoTxIn addr
+
+    let dummyOut = TxOut addr (Coin 10000000000)
+
+    let cs = CoinSelection
+            { inputs = [(txIn, dummyOut)]
+            , withdrawal = 0
+            , reclaim = 0
+            , outputs = []
+            , change = []
+            , deposit = 100000 -- keyDeposit
+            }
```

Comment:
CoinSelection is a `Monoid`, so you can write instead: 

```hs
mempty { inputs = [(txIn, dummyOut)], deposit = 100000 }
```

Also, note that the key deposit is available in protocol parameters, would be better than hard-coding it.

Reply:
> key deposit is available in protocol parameters
👍 

> CoinSelection is a Monoid
Didn't think about it, but in this case I found it useful to explicitly see the new fields of `CoinSelection`.